### DISCUSSION TITLE
Developer documentation

### DISCUSSION THREAD

**Discussion by Darylgolden** [ManimCommunity/manim#1717] [Suggestions and Proposals]
Having proper developer documentation would help new contributors (and even experienced devs) understand and contribute to the library. Should we somehow merge it with the current documentation, or have it as a separate thing? If we merge it, how should we do so?

**Comment by behackl** [comment]
I'm not quite sure I get the idea. What would the developer documentation consist of?

**Reply by WampyCakes** [reply]
I would say internals like how `play` works or scene caching or how the commandline stuff works (maybe a bit less needed since the refactor, but we still have to know how to add flags properly if this hasn't already been re-documented). Things of that nature

**Comment by huguesdevimeux** [comment]
I'm very, very, very in favor of this. I remember how painful it was (and still is :')) to get a comprehensive sight of the internals wheels of manim, and I remember how much I'd have blessed some dev docs. 

The idea would be to give everything useful so a new contributor is able to have a global understanding of how manim works under the hood. 
I think the first thing to write should be what happen when you use `.play`, because all the scene writing thing is far from being obvious.
Next, we should write something about VMobjects and Bézier curves, how they are rendered (with cairo), since this is as well a complicated topic, and the code is complex.
Next, the Config system, the caching, the tex handling, etc. 

In other terms, everything that is manim internal related, but can't fit it doc string because it's too dev specific should go there. 

If anyone is interested in doing this, please let it know so we can go further !

**Comment by k4pran** [comment]
As mentioned on discord I’m also keen on this idea! It seems to me that once we get started on it it would be very much a community effort as different devs have areas of the code they are more knowledgeable in.

**Reply by huguesdevimeux** [reply]
Yes, we just need someone to just start. I'm sure that if there is a structure already existing, a lot of devs (including myself) will be happy to write stuff.

(I can't start atm)
